routers/web/repo/release.go

```diff
@@ -207,6 +214,12 @@ func TagsList(ctx *context.Context) {
 	ctx.Data["HideBranchesInDropdown"] = true
 	ctx.Data["CanCreateRelease"] = ctx.Repo.CanWrite(unit.TypeReleases) && !ctx.Repo.Repository.IsArchived
 
+	if !ctx.Repo.CanRead(unit.TypeReleases) && ctx.Repo.CanRead(unit.TypeCode) {
+		if !shared.RenderRepoSubMenu(ctx) {
+			return
+		}
+	}
```

Comment:
The change is good but I really dislike the unclear code.

* Why "render"? What does it really "render"? It only prepares some data for the template?
* Why "RepoSubMenu" could be used in this PR? It only prepares license data but not others?
* Why duplicate code to check permissions? Why not check permissions in the function.
* Why `!CanRead(release) && CanRead(code)`? Does it really make sense?

Reply:
d61ac5dd52 changed the function names and added some comments.